src/com/itacademy/eshop/shop/ShoppingCart.java

```diff
@@ -26,4 +26,22 @@ public String getTotalPrice() {
          */
         throw new UnsupportedOperationException("getTotalPrice() is not implemented yet");
     }
+
+    public void addProduct(Product shirt) {
+        products.add(shirt);
+    }
+
+    public void removeProductByName(String laptop) {
```

Comment:
the argument passed to this method should have a more generic name, as you can pass values other than "laptop" to it